rust/main/chains/hyperlane-cosmos/src/providers/cosmos/provider.rs

```diff
@@ -103,6 +110,46 @@ impl CosmosProvider {
         Ok((account_id, signer_info.sequence))
     }
 
+    fn normalize_public_key(
+        &self,
+        signer_public_key: SignerPublicKey,
+    ) -> ChainResult<SignerPublicKey> {
+        let public_key = match signer_public_key {
+            SignerPublicKey::Single(pk) => SignerPublicKey::from(pk),
+            SignerPublicKey::LegacyAminoMultisig(pk) => SignerPublicKey::from(pk),
+            SignerPublicKey::Any(pk) => {
+                if pk.type_url != INJECTIVE_PUBLIC_KEY_TYPE_URL {
+                    let msg = "we don't support this kind of TYPE_URL for public key";
```

Comment:
nit: can you make this message a bit more specific, like "can only normalize public keys with a known TYPE_URL" or similar

Reply:
Good point. Thank you!